Added changes from Linear ticket #CF-3659

VigneshSelvaraj96 opened the pull request:
Fixed version numbers for terraform providers and added cedar policies according to changes highlighted in Linear ticket: https://linear.app/common-fate/issue/CF-3659/fixes-to-byoc-starter-config

What changed?
Updates CF terraform provider's version numbers and added extra cedar policies for control plane.

Why?
Older CF terraform provider requires an extra argument in other resources that are not documented in config docs. Control plane Cedar policy was required for deployment.

How did you test it?
Tested locally with my own byoc @vignesh.lightwave.dev

Potential risks
low-medium, this is just a starter config that only new teams who might use in the future might have issues with. Local testing does not seem to break anything.

Is patch release candidate?
N/A

Link to relevant docs PRs
N/A

VigneshSelvaraj96 committed:
Added changes from Linear ticket https://linear.app/common-fate/issue/CF-3659/fixes-to-byoc-starter-config

chrnorm commented on config/policies/default.cedar:
```diff
@@ -44,4 +44,21 @@ forbid (
+
+// Allow admins to read all resources
+permit (
+    principal,
```
the policy here is a bit too permissive as it allows everyone to have admin read access, let's scope it down to suggest to the user they should do the same
```suggestion
    principal in Entra::Group::"ID_OF_COMMON_FATE_ADMINISTRATOR_GROUP",
```

chrnorm commented on config/policies/default.cedar:
```diff
@@ -44,4 +44,21 @@ forbid (
+);
+
+
+permit (
+    principal ==
+        CF::User::<Replace with your own user id from CF web console>,
+    action ==
+        CF::RPC::Action::"commonfate.control.integration.v1alpha1.IntegrationService/ListSyncsForIntegration",
+    resource ==
+        CF::Service::"ControlPlane",
+);
```
This can be removed, in new versions the ListSyncsForIntegration action is correctly grouped in the `CF::Admin::Action::"Read"` group

```suggestion
```

VigneshSelvaraj96 committed:
Update config/policies/default.cedar

Removed Control Plane actions

Co-authored-by: Chris Norman <17420369+chrnorm@users.noreply.github.com>

VigneshSelvaraj96 committed:
Update config/policies/default.cedar

Updated permission so that it's less permissive (previously allowed everyone to have admin read access)

Co-authored-by: Chris Norman <17420369+chrnorm@users.noreply.github.com>
